lib/practice_exercises.rb

```diff
@@ -1,13 +1,78 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) n being all elements in the list. They are get iterated through
+#compared. 
+# Space Complexity: ?Would be O(1) the space stays the same throughout. No new arrays 
+# are created
+
+#compare the first number to all numbers until you reach a num thats not equal
+#Then replace taht first num with num that is different, and add to the counter
+#Once you reach the end of the array return the counter.
+
+
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  if list.length == nil
+    return []
+  elsif list[0] == []
+    return []
+  end
+  
+  compare_to = list[0]
+  i = 0
+  counter = 0
+  
+  while list.length >= i
+    if compare_to == list[i]
+      list.delete_at(list[i])
```

Comment:
`delete_at` takes an index number as an argument **not** an element.  So that's why this isn't working. 

`delete_at` also shifts each subsequent element over one index so it's an O(n) method.  Since it's nested inside a loop which runs n times, this means the method would be O(n<sup>2</sup>)

Also if you delete an element you should decrement `i` by one so that you don't end up skipping any duplicates.